Mapping.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef CONTROLDEV_WEBSOCKET_TYPES_HPP
+#define CONTROLDEV_WEBSOCKET_TYPES_HPP
+
+#include <string>
+
+namespace controldev_websocket
+{
+    struct Mapping
+    {
+        enum Type {
+            Axis = 0,
+            Button = 1
+        };
+
+        /**
+         * Enum to represent an axis or a button.
+         *
+         *
+         * @note don't use this value directly. It's only public to allow this class
+         * to be used as an interface type.
+         */
+        Type type;
+
+        std::string getType(){
```

Comment:
`getXXX` should really only return the value from the corresponding field. The goal of this method is to return the JSON field that matches this mapping, so getJSONFieldName() would be a better name. In addition, a plain getter like this should be const.

Moreover, I think it would be better to not "leak" internal stuff like this in the generic type.

Reply:
I don't agree, the interface type is to do this. Ideally, if I change the browser side I should just update Mapping (except if I change status). For status maybe it is good to put it as a parameter, although it should not be changing it will be good as doc.

Reply:
This method is not an intrinsic operation of the mapping type. It is a function that makes the link between the type and the representation of the data structure in JSON, which is an operation specific to the task. Hence, I don't think it should be here.